src/model/decorator.ts

```diff
@@ -0,0 +1,8 @@
+import "reflect-metadata";
```

Comment:
It is for defining and reading metadata on types. I use it to support `Serialize` decorator so that every class can customize which property to be serialized.

Reply:
I see, nonetheless, if `TypedJSON` can satisfy our need, it might be more robust than handling the serialization/deserialization on our own.